src/librustc_typeck/astconv.rs

```diff
@@ -409,6 +411,8 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                 arg_counts.types,
                 arg_counts.lifetimes,
             )
+        } else if let Some(is_err) = reported_late_bound_region_err {
```

Comment:
this could be merged into the `else` below by using the same `reported_late_bound_region_err.unwrap_or(false)` as in the early return further up

r=me with this nit

Reply:
Fixed, thanks.